### ISSUE TITLE
[Bug]: Load collection slowly, segment DMLPosition update failed

### ISSUE THREAD

**Issue by bigsheeper** [milvus-io/milvus#19733]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.1.2-3c3ba55c
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2):github.com/milvus-io/milvus-sdk-go/v2 v2.1.2-0.20220823032848-8832edc6f96e
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

[autohome issue]
Some segment's DMLPositions are delayed for days. So when milvus load collection, querynode took a long time to `FromDMLCOLoadDelete`.  5million rows of data took hours to load which is abnormal.
![image](https://user-images.githubusercontent.com/42060877/195350781-0c9f43d6-f7b8-4954-83bf-5fc956389957.png)

### Anything else?

Collection info:
![image](https://user-images.githubusercontent.com/42060877/195097637-123da329-96ed-4f63-8d5b-3a3829ecc425.png)

Segments info:
![image](https://user-images.githubusercontent.com/42060877/195097846-764e88b2-982d-4760-9e10-048db9094508.png)

**Comment by bigsheeper** [comment]
Collection checkpoint and abnormal segment info:
![image](https://user-images.githubusercontent.com/42060877/195344265-06d76252-9207-4020-99a5-b42afe2b3916.png)

**Comment by bigsheeper** [comment]
[bw_etcd_ALL.221010-210929.bak.gz](https://github.com/milvus-io/milvus/files/9764950/bw_etcd_ALL.221010-210929.bak.gz)

**Comment by bigsheeper** [comment]
[datacoord-59-2022-10-10T21-20-41.044.log.zip](https://github.com/milvus-io/milvus/files/9765013/datacoord-59-2022-10-10T21-20-41.044.log.zip)
[querycoord-4.log.zip](https://github.com/milvus-io/milvus/files/9765018/querycoord-4.log.zip)

[querynode-72.log.zip](https://github.com/milvus-io/milvus/files/9765022/querynode-72.log.zip)

[querynode-78.log.zip](https://github.com/milvus-io/milvus/files/9765024/querynode-78.log.zip)

[datanode61.log.tar.gz](https://github.com/milvus-io/milvus/files/9765034/datanode61.log.tar.gz)

**Comment by soothing-rain** [comment]
Happened when compaction fails in DataCoord but succeeded in DataNode. DataNode was not aware that the compaction failure and proceed to updated its meta.

So basically DataNode will remove this segment from its flowgraph:

```
[2022/10/08 23:16:31.692 +08:00] [DEBUG] [datanode/flow_graph_delete_node.go:230] ["update delBuf for compacted segments"] ["compactedTo segmentID"=436532988063252481] ["compactedFrom segmentIDs"="[436424709483790339]"]
[2022/10/08 23:16:31.692 +08:00] [INFO] [datanode/segment_replica.go:647] ["remove segments if exist"] [segmentIDs="[436424709483790339]"]
```

All following flush from DataCoord would fail because the segment was not longer in flowgraph.

```
[2022/10/08 23:18:16.493 +08:00] [INFO] [datanode/data_node.go:573] ["receiving FlushSegments request"] ["collection ID"=436120504361287681] [segments="[]"] ["stale segments"="[436424709483790339]"]
[2022/10/08 23:18:16.493 +08:00] [INFO] [datanode/data_node.go:617] ["flow graph flushSegment tasks triggered"] [flushed=true] ["collection ID"=436120504361287681] ["segments sending to flush channel"="[]"]
[2022/10/08 23:18:16.493 +08:00] [ERROR] [datanode/data_node.go:594] ["no flush channel found for the segment, unable to flush"] ["segment ID"=436424709483790339] [error="cannot find segment 436424709483790339 in all flowgraphs"] [stack="github.com/milvus-io/milvus/internal/datanode.(*DataNode).FlushSegments.func1\n\t/go/src/github.com/milvus-io/milvus/internal/datanode/data_node.go:594\ngithub.com/milvus-io/milvus/internal/datanode.(*DataNode).FlushSegments\n\t/go/src/github.com/milvus-io/milvus/internal/datanode/data_node.go:625\ngithub.com/milvus-io/milvus/internal/distributed/datanode.(*Server).FlushSegments\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/datanode/service.go:345\ngithub.com/milvus-io/milvus/internal/proto/datapb._DataNode_FlushSegments_Handler.func1\n\t/go/src/github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:5825\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/datapb._DataNode_FlushSegments_Handler\n\t/go/src/github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:5827\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:1283\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:1620\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:922"]
```

Luckily this issue is resolved in 2.1.4 :)

https://github.com/milvus-io/milvus/pull/19476

/unassign @congqixia 
/assign @yanliang567

**Comment by yanliang567** [comment]
close for now, will check in benchmark
